### PR TITLE
Added DIY flags for items and DIY popup for lists

### PR DESCRIPTION
**Pull request by trundleroo**
Still a bit of a work in progress. The items need to show a DIY checkbox instead of a dropdown entry.

### CODE REVIEW

File: db/entity/items.js [hunk 1]
```diff
@@ -255,12 +255,16 @@ class Items extends RedisStore {
...
+                if (Object.keys(game.variations).length > 1) {
+                    Object.assign(variations, game.variations);
+                }
             }
         }
 
-        // Only allow variations selection if more than one variant exists.
-        if (Object.keys(variations).length < 2) {
+        // Add DIY variation for items that have a recipe.
+        if (Object.keys(item.games).includes('nh')) {
```
Comment (jefflomacy) [inline db/entity/items.js:265]: can just do if (item.games.nh)

File: db/entity/items.js [hunk 2]
```diff
@@ -299,6 +303,16 @@ class Items extends RedisStore {
             item.variationImages[k] = imageData;
         }
     }
+
```
Comment (jefflomacy) [inline db/entity/items.js:306]: function comment /**

File: db/entity/items.js [hunk 3]
```diff
@@ -299,6 +303,16 @@ class Items extends RedisStore {
...
+    diyChecks(variations, item) {
+        if (Object.keys(item.games.nh).includes('recipe')) {
```
Comment (jefflomacy) [inline db/entity/items.js:308]: if (item.games.nh.recipe)

File: db/entity/items.js [hunk 4]
```diff
@@ -255,12 +255,16 @@ class Items extends RedisStore {
         for (let gameId in item.games) {
             const game = item.games[gameId];
             if (typeof game.variations !== 'undefined') {
-                Object.assign(variations, game.variations);
+                if (Object.keys(game.variations).length > 1) {
```
Comment (jefflomacy) [inline db/entity/items.js:258]: what was the reasoning for changing this from just the singular assign statement?

File: db/entity/items.js [hunk 5]
```diff
@@ -299,6 +303,16 @@ class Items extends RedisStore {
             item.variationImages[k] = imageData;
         }
     }
+
+    diyChecks(variations, item) {
+        if (Object.keys(item.games.nh).includes('recipe')) {
+            if (Object.keys(item.games.nh.recipe).length > 0) {
+                console.log('Adding DIY entry for ' + item.id);
+                const diy = {'_isDIY': 'DIY'};
+                Object.assign(variations, diy);
```
Comment (jefflomacy) [inline db/entity/items.js:312]: you can change tehse two lines to just variations['_isDIY'] = 'DIY';

File: routes/user.js [hunk 6]
```diff
@@ -153,6 +153,11 @@ function organizeData(listId, entity, type, variationId) {
         entityData.deleteUrl += '/' + variationId;
         entityData._sortKey += '-vv-' + variationId;
 
+        if (entityData.variationId == '_isDIY') {
```
Comment (jefflomacy) [inline routes/user.js:156]: === for type safety, and let's define '_isDIY' somewhere as a constant that is exported from a module

File: db/entity/items.js (hunk 4 above)
Comment (trundleroo) [reply db/entity/items.js:258]: The only thing I added here is an if statement to ensure items with more than 1 variation are assigned their variations. This allows for the DIY variation to be added to items with no variations without the method returning.

**Pull request merged**